src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -537,14 +537,24 @@ class DocBaseViewer extends BaseViewer {
      */
     setPage(pageNumber) {
         const parsedPageNumber = parseInt(pageNumber, 10);
-        if (!parsedPageNumber || parsedPageNumber < 1 || parsedPageNumber > this.pdfViewer.pagesCount) {
+        // if the preloader has opened the thumbnails and the pdf has not finished loading go to the doc first page
+        // for that thumbnail
+        if (this.thumbnailsSidebar && this.preloader?.thumbnailsOpen && this.pdfViewer.pagesCount === 0) {
+            this.cachePage(parsedPageNumber);
```

Comment:
What does this do?

Reply:
it stores the page you are on in local storage so that the next time you come to the document it takes you where you left off.